modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -557,6 +557,10 @@ public abstract class IgniteUtils {
     private static final DateTimeFormatter THREAD_DUMP_FMT =
         DateTimeFormatter.ofPattern("yyyy/MM/dd HH:mm:ss z").withZone(ZoneId.systemDefault());
 
+    /** ISO-8601 compliant date-time format. */
+    public static final DateTimeFormatter ISO_DATE_FMT =
+        DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss,SSS").withZone(ZoneId.systemDefault());
```

Comment:
can we use `DateTimeFormatter.ISO_LOCAL_DATE_TIME` instead?

Reply:
I thought about it also, but ISO_LOCAL_DATE_TIME doesn't print millis

Reply:
Why do we need millis, since ISO_LOCAL_DATE_TIME prints nanoseconds?

Reply:
Are you sure about nanos? From ISO_LOCAL_DATE_TIME javadoc:
The ISO date-time formatter that formats or parses a date-time without an offset, such as '2011-12-03T10:15:30'.

Anyway, millis are useful when you need to understand what happened first (there can be too many events in during whole second), but nanos would be an overkill.

Moreover proposed format yyyy-MM-dd'T'HH:mm:ss,SSS matches our default format used by ignite-log4j2:
<PatternLayout pattern="[%d{ISO8601}][%-5p][%t][%c{1}]%notEmpty{[%markerSimpleName]} %m%n"/>

2023-05-08T09:04:06,626 INFO  [pgrunner-#node-1] [o.a.i.i.IgniteKernal%node-1] Logger: Log4J2Logger

Reply:
> Are you sure about nanos?

yes, because it uses `ISO_LOCAL_TIME` internally, which states `One to nine digits for the nano-of-second. As many digits will be output as required`

Anyway, I won't insist, current approach is also fine, so it's up to you